src/routing/approved.rs

```diff
@@ -1449,6 +1452,10 @@ impl Approved {
             self.send_event(Event::Demoted);
         }
 
+        if self.node.age > MIN_AGE && !self.is_elder() {
```

Comment:
Now this is going to be called every time there is a section change, not just the first time the node goes from `MIN_AGE` to `> MIN_AGE`. That's probably not what we want, is it?

Reply:
Yea, that and `update_state` function triggers only when an event comes by, so yea this might not be the right place to rise this event. And we need to fire it when we get `Demoted` from being an Elder too. Node should become an adult when demoted from an Elder. So sn_node should receive the `PromotedToAdult` event in both cases.